core/types/calls.cc

```diff
@@ -499,7 +499,13 @@ DispatchResult dispatchCallSymbol(const GlobalState &gs, DispatchArgs args,
             return DispatchResult(Types::untypedUntracked(), std::move(args.selfType), Symbols::untyped());
         }
         auto result = DispatchResult(Types::untypedUntracked(), std::move(args.selfType), Symbols::noSymbol());
-        if (auto e = gs.beginError(args.locs.call, errors::Infer::UnknownMethod)) {
+        // We want to always be able to build the error object
```

Comment:
```suggestion
        // This is a hack. We want to always be able to build the error object
```

Reply:
Note that the same applies to argument adaptation, right?
Happy to have it done either in this PR or another one

Reply:
Hmm, I am not too familiar with the argument adaptation part. Would love to take a look if you show me the way, though.